segmentation_models_pytorch/unet3plus/decoder.py

```diff
@@ -0,0 +1,245 @@
+          elif scale==0:
+            block = ConstBlock(in_channels[i], out_channels[output_index],0, **kwargs)
+          else:
+            block = UpBlock(previous_channels[output_index-prev_ind], out_channels[output_index], pow(2, abs(scale)), **kwargs)
+            prev_ind+=1
+
+          blocks.append(block)
+          scale = scale-1
+        
+        self.blocks = nn.ModuleList(blocks)
+
+        #concatenation block
+        self.cat_block = md.Conv2dReLU(
+            self.UpChannel,
+            self.UpChannel,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm,
+        )
+        self.attention_cat = md.Attention(attention_type, in_channels=self.UpChannel)
+
+    def forward(self, feature):
+        
+        result_list = []
+        for i, block in enumerate(self.blocks):
+          result = block(feature[i])
+          result_list.append(result)
+
+        concat_tensor = torch.cat(result_list, 1) #concatenation of every tensor
+        final = self.cat_block(concat_tensor)
+        final = self.attention_cat(final)
+
+        return final
+
+
+class CenterBlock(nn.Sequential):
+    def __init__(self, in_channels, out_channels, use_batchnorm=True):
+        conv1 = md.Conv2dReLU(
+            in_channels,
+            out_channels,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm,
+        )
+        conv2 = md.Conv2dReLU(
+            out_channels,
+            out_channels,
+            kernel_size=3,
+            padding=1,
+            use_batchnorm=use_batchnorm,
+        )
+        super().__init__(conv1, conv2)
+
+
+class UnetDecoder(nn.Module):
+    def __init__(
+            self,
+            encoder_channels,
+            decoder_channels,
+            n_blocks=5,
+            use_batchnorm=True,
+            attention_type=None,
+            center=True,
+    ):
+        super().__init__()
+
+        """if n_blocks != len(decoder_channels):
+            raise ValueError(
+                "Model depth is {}, but you provide `decoder_channels` for {} blocks.".format(
+                    n_blocks, len(decoder_channels)
+                )
+            )"""
+
+        encoder_channels = encoder_channels[1:]  # remove first skip with same spatial resolution
+        encoder_channels = encoder_channels[::-1]  # reverse channels to start from head of encoder
+
+        # computing blocks input and output channels
+        head_channels = encoder_channels[0]
+        in_channels = encoder_channels[1:]
+        in_channels = in_channels[::-1] #reversing the list
+
+        out_channels = [head_channels] + list(decoder_channels) 
+        previous_channels = [i * n_blocks for i in out_channels] #calculating previos channels i.e concatenated output channels of previous block
+        previous_channels[0] = int(previous_channels[0]/n_blocks) #first tensor as it comes from CenterBlock, hence no concatenation
+
+        if True:
```

Comment:
Looks strange: `else` statement for `if True`

Reply:
Yeah, it should be center. I will update that